From d983c89dbce840fad50f49e4253ecc7930f15338 Mon Sep 17 00:00:00 2001
From: Chris Friesen <chris.friesen@windriver.com>
Date: Wed, 6 Feb 2019 17:19:39 -0600
Subject: [PATCH] fix type error to streamline single-replica mariadb startup

The mariadb startup script was trying to optimize the single-replica
case but missed the fact that the variable it was checking was a
string rather than an int.

Converting it to an int before doing the comparison makes it work
as expected.

Change-Id: I8612e9e8ef5ec8ff61ecf0782f262a5feafd501a
Signed-off-by: Chris Friesen <chris.friesen@windriver.com>
---
 mariadb/templates/bin/_start.py.tpl | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mariadb/templates/bin/_start.py.tpl b/mariadb/templates/bin/_start.py.tpl
index 4d4428c..d0b9c8e 100644
--- a/mariadb/templates/bin/_start.py.tpl
+++ b/mariadb/templates/bin/_start.py.tpl
@@ -729,7 +729,7 @@ elif get_cluster_state() == 'init':
     run_mysqld()
 elif get_cluster_state() == 'live':
     logger.info("Cluster has been running starting restore/rejoin")
-    if not mariadb_replicas > 1:
+    if not int(mariadb_replicas) > 1:
         logger.info(
             "There is only a single node in this cluster, we are good to go")
         update_grastate_on_restart()
-- 
1.8.3.1

